File: 0643-maximum-average-subarray-i/0643-maximum-average-subarray-i.java
class Solution {
    public double findMaxAverage(int[] nums, int k) {
        
        int cnt = 0;
        double sum = 0;
        double result = Integer.MIN_VALUE;

        for(int i = 0; i<nums.length; i++){
            if(cnt<k){
                sum+=nums[i];
                cnt++;
            }
            else{
                sum-=nums[i-k];
                sum+=nums[i];
            }
            System.out.println(result+" "+sum/k);
            result = Math.max(result, sum/k);
        }

        return result;
    }
}

File: 0643-maximum-average-subarray-i/0643-maximum-average-subarray-i.py
class Solution:
    def findMaxAverage(self, nums: List[int], k: int) -> float:
        result = 0
        curr = 0
        for i in nums[:k]:
            curr += i
        result = curr

        for i in range(k, len(nums)):
            curr += nums[i]
            curr -= nums[i-k]

            result = max(result, curr)
        
        return result/k
